### PR TITLE
Subscription activation and registry service account configuration using available REST APIs

### LINKED ISSUE #30: Implement subscription management

### PR DESCRIPTION
**Pull request by dgolovin**
Fix #30, #25

### CODE REVIEW

**Comment by vrothberg** [comment]
Can we merge this PR into the `main` branch?  I think this will make things simpler as interested parties may otherwise forget about the other branch.

**Comment by jeffmaury** [comment]
The main branch is used to deliver fixes for the released extension. We should not merge this in main unless we are sure that existing code using this (Sandbox, OpenShift Local) are not being degraded

**Comment by benoitf** [comment]
@jeffmaury maybe we do a branch for bugfixes ? (branches = bugfixes, main = development as it might be more expected)

I left comments on the PR https://github.com/redhat-developer/podman-desktop-redhat-account-ext/pull/26 before it has been recreated (don't know why ? ) that are not yet addressed

**Comment by jeffmaury** [comment]
> @jeffmaury maybe we do a branch for bugfixes ? (branches = bugfixes, main = development as it might be more expected)
I'm fine
> 
> I left comments on the PR #26 before it has been recreated (don't know why ? ) that are not yet addressed

**Comment by vrothberg** [comment]
That sounds like a good way forward to me as well.  The (README of the) main branch is what users will hit first.

File: src/podman-cli.ts
Comment (dgolovin) [inline src/podman-cli.ts:20]: > @benoitf: if it needs to stay even after podman 5.0 I think you should not have duplicates of the podman extension code > there
>
> probably we would need to see how podman extension can export an API that this extension could consume

It can be new, part of provider or CLI API.

@benoitf WDYT?

File: tsconfig.json
Comment (dgolovin) [inline tsconfig.json:14]: > @benoitf: I think all github workflows, dependency updates, tsconfig changes could be separated small PR as they're not related to the changes

I left only required changes to fix build errors.

**Comment by dgolovin** [comment]
@benoitf I renamed to match the release version. All requested changes are in.

**Comment by dgolovin** [comment]
> @dgolovin  That sounds like a good way forward to me as well. The (README of the) main branch is what users will hit first.

I do like main for development, branch for fixes approach.

**Comment by vrothberg** [comment]
One suggestion from my end:
```diff
diff --git a/www/success.html b/www/success.html
index 49f6aa9239b2..500944c7fd8a 100644
--- a/www/success.html
+++ b/www/success.html
@@ -2,7 +2,7 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <title>Red Hat OpenShift Account Services</title>
+    <title>Red Hat Account Login</title>
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <link rel="stylesheet" type="text/css" media="screen" href="auth.css" />
     <link
@@ -20,7 +20,7 @@
           Welcome to <span id="service">Red Hat</span>
         </h1>
         <div class="pf-c-empty-state__body message">
-          You have successfully logged as
+          You have successfully logged in as
           <span style="font-weight: 700" id="login"></span>. <br />
           You may now close this page.
         </div>
```

File: package.json
Comment (dgolovin) [inline package.json:5]:
```suggestion
  "version": "0.0.2-next",
```

File: Containerfile
Comment (dgolovin) [inline Containerfile:23]:
```suggestion
  io.podman-desktop.api.version=">= 1.7.0"
```

File: src/extension.ts
Comment (dgolovin) [inline src/extension.ts:70]:
```suggestion

const REGISTRY_REDHAT_IO = 'registry.redhat.io';

```

File: src/extension.ts
Comment (dgolovin) [inline src/extension.ts:71]:
```suggestion
async function createRegistry(username: string, secret: string, serverUrl: string = REGISTRY_REDHAT_IO): Promise<void> {
```

File: src/extension.ts
Comment (dgolovin) [inline src/extension.ts:97]:
```suggestion
    configured =  authFileJson?.auths?.hasOwnProperty(REGISTRY_REDHAT_IO) || false;
```

File: src/extension.ts
Comment (dgolovin) [reply src/extension.ts:116]: It is name of the service account and activation key to make it clear where those are being used.

File: src/configuration.ts
Comment (dgolovin) [inline src/configuration.ts:39]: This is the client ID in SSO

File: src/extension.ts
Comment (dgolovin) [inline src/extension.ts:86]:
```suggestion
  if (!homeFolderPath) {
    throw new Error('Unable to find home directory for the current user');
  }
  const pathToAuthJson = path.join(homeFolderPath, '.config', 'containers', 'auth.json');
```

File: src/extension.ts
Comment (dgolovin) [reply src/extension.ts:122]: Created follow up issue https://github.com/redhat-developer/podman-desktop-redhat-account-ext/issues/43

File: src/extension.ts
Comment (dgolovin) [reply src/extension.ts:146]: Created follow up issue https://github.com/redhat-developer/podman-desktop-redhat-account-ext/issues/43

File: src/extension.ts
Comment (dgolovin) [reply src/extension.ts:138]: it always returns the session or throws exception.

**Comment by dgolovin** [comment]
> @vrothberg One suggestion from my end:

updated

File: src/extension.ts
Comment (dgolovin) [reply src/extension.ts:185]: I added console.error calls in podman-cli.ts to report commands being executed and error codes

File: src/podman-cli.ts
Comment (benoitf) [inline src/podman-cli.ts:118]: looks like there is a missing EOF (not blocking)

**Review by benoitf — APPROVED** [review]
can use follow-ups 👍

**Pull request merged**
